### ISSUE TITLE
Error on parent property when class is extended

### ISSUE THREAD

**Issue by talmog** [palantir/tslint#1066]
### Bug Report (delete this for feature requests)

- __TSLint version__: `3.5.0`
- __Running TSLint via__: gulp ts-line version 4.3.3

#### TypeScript code being linted

```ts
export class Child extends Base  {
  public property1;
}

export class Base  {
  public property2;
}

let child = new Child();
child.property2 = ' whatever';
```

#### Actual behavior

error: property 'property2' does not exist on type 'Base'.

#### Expected behavior

No error

**Comment by adidahiya** [comment]
This is a compiler error, not a linter error.

**Comment by talmog** [comment]
But I am getting a lint error! Also, why is this a complier error?
